core/src/impl/Kokkos_ViewLayoutTiled.hpp

```diff
@@ -1142,216 +1142,89 @@ struct ViewMapping<
-      SrcLayout;
+}  // end namespace _tile_subview_helpers
+}  // end namespace Impl
 
-  return Kokkos::View<T[N0][N1][N2][N3][N4][N5][N6], array_layout, P...>(
-      src, SrcLayout(), i_tile0, i_tile1, i_tile2, i_tile3, i_tile4, i_tile5,
-      i_tile6);
-}
+//----------------------------------------------------------------------------
+//----------------------------------------------------------------------------
 
-// Rank 8
-template <typename T, Kokkos::Iterate OuterP, Kokkos::Iterate InnerP,
+// Rank N
+template <typename U, Kokkos::Iterate OuterP, Kokkos::Iterate InnerP,
           unsigned N0, unsigned N1, unsigned N2, unsigned N3, unsigned N4,
-          unsigned N5, unsigned N6, unsigned N7, class... P>
-KOKKOS_INLINE_FUNCTION Kokkos::View<
-    T[N0][N1][N2][N3][N4][N5][N6][N7],
+          unsigned N5, unsigned N6, unsigned N7, class... P, class... Tiles>
+KOKKOS_INLINE_FUNCTION
+Kokkos::View<
+    typename Impl::_tile_subview_helpers::_compose_array<
+        sizeof...(Tiles),
+        typename Impl::_tile_subview_helpers::_element_type<U>::type, N0, N1,
+        N2, N3, N4, N5, N6, N7>::type,
     typename std::conditional<(InnerP == Kokkos::Iterate::Left),
                               Kokkos::LayoutLeft, Kokkos::LayoutRight>::type,
     P...>
 tile_subview(const Kokkos::View<
-                 T********,
+                 U,
                  Kokkos::Experimental::LayoutTiled<OuterP, InnerP, N0, N1, N2,
                                                    N3, N4, N5, N6, N7, true>,
                  P...>& src,
-             const size_t i_tile0, const size_t i_tile1, const size_t i_tile2,
-             const size_t i_tile3, const size_t i_tile4, const size_t i_tile5,
-             const size_t i_tile6, const size_t i_tile7) {
+             Tiles const... i_tiles) {
   // Force the specialized ViewMapping for extracting a tile
   // by using the first subview argument as the layout.
-  typedef
-      typename std::conditional<(InnerP == Kokkos::Iterate::Left),
-                                Kokkos::LayoutLeft, Kokkos::LayoutRight>::type
-          array_layout;
-  typedef Kokkos::Experimental::LayoutTiled<OuterP, InnerP, N0, N1, N2, N3, N4,
-                                            N5, N6, N7, true>
-      SrcLayout;
+  using src_view_type =
+      Kokkos::View<U,
+                   Kokkos::Experimental::LayoutTiled<OuterP, InnerP, N0, N1, N2,
+                                                     N3, N4, N5, N6, N7, true>,
+                   P...>;
+  using src_layout = typename src_view_type::array_layout;
+  static_assert(sizeof...(i_tiles) == src_view_type::rank,
+                "Mismatched tile/array size");
 
-  return Kokkos::View<T[N0][N1][N2][N3][N4][N5][N6][N7], array_layout, P...>(
-      src, SrcLayout(), i_tile0, i_tile1, i_tile2, i_tile3, i_tile4, i_tile5,
-      i_tile6, i_tile7);
+  return { src, src_layout(), static_cast<size_t>(i_tiles)... };
 }
 
 } /* namespace Kokkos */
```

Comment:
What is all this? Does this have anything to do with the rest of the PR or is it just cleanup of ViewLayoutTiled?